### PR TITLE
test the result of some more allocation functions

### PR DESCRIPTION
**Pull request by pascal-cuoq**

### CODE REVIEW

**Comment by lexborisov** [comment]
Thanks again!

File: source/myhtml/token.c [hunk 1]
```diff
@@ -103,9 +103,19 @@ myhtml_token_t * myhtml_token_create(myhtml_tree_t* tree, size_t size)
         size = 4096;
     
     myhtml_token_t* token = (myhtml_token_t*)myhtml_malloc(sizeof(myhtml_token_t));
-    
+    if (!token) return token;
+
     token->nodes_obj = mcobject_async_create();
+    if (!token->nodes_obj) {
+        free(token);
```
Comment (f2404) [inline source/myhtml/token.c]: Shouldn't `myhtml_free` be used here?

File: source/myhtml/token.c [hunk 2]
```diff
@@ -103,9 +103,19 @@ myhtml_token_t * myhtml_token_create(myhtml_tree_t* tree, size_t size)
...
+        return NULL;
+    }
     token->attr_obj  = mcobject_async_create();
+    if (!token->attr_obj) {
+        free(token->attr_obj);
+        free(token);
```
Comment (f2404) [inline source/myhtml/token.c]: And here as well.

File: source/myhtml/token.c [hunk 3]
```diff
@@ -103,9 +103,19 @@ myhtml_token_t * myhtml_token_create(myhtml_tree_t* tree, size_t size)
         size = 4096;
     
     myhtml_token_t* token = (myhtml_token_t*)myhtml_malloc(sizeof(myhtml_token_t));
-    
+    if (!token) return token;
```
Comment (f2404) [inline source/myhtml/token.c]: Suggest replacing `if (!smth)` with `if (smth == NULL)` - similar to other code.

**Comment by lexborisov** [comment]
Thanks, Igor! Immediately I noticed. I just can not figure out whether it is possible to change pull request here before merge?

**Comment by f2404** [comment]
I don't think you can change another person's PR from here.

One option could be to merge PR as is to respect the contributor's work, and add your fixes on top of it in another commit.
Another option could be to download the contributor's patch, fix it and commit with `--author` option - thus specifying the contributor as the commit author.

**Comment by lexborisov** [comment]
Ok. Thank you!

**Comment by f2404** [comment]
@lexborisov You're welcome.
Congrats with the release, btw! :) I've seen the article on opennet, and it's got many comments. Keep it up!

**Comment by lexborisov** [comment]
@f2404 Thank you! And thanks for the help in the project! I try!

**Pull request merged**